File: BackEnd/src/main/java/imi/spring/backend/models/PostDTO.java
package imi.spring.backend.models;

import imi.spring.backend.models.mongo.Photo;
import imi.spring.backend.models.mongo.Video;
import lombok.Data;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class PostDTO {

    private Long appUserId;
    private String appUserUsername;

    private Long postId;
    private Location location;
    private String description;
    private Integer numberOfLikes;
    private Integer numberOfComments;
    private Boolean isLiked; //da li je logovani user lajkovao ovaj post

//    @JsonIgnore
    private List<Photo> photos;
    @JsonIgnore
    private List<Video> videos;

    private String image;

    private String date;
}

File: FrontEnd/app/src/main/java/com/example/frontend/data/remote/dto/PostDTO.kt
package com.example.frontend.data.remote.dto

import com.example.frontend.domain.model.Location
import com.example.frontend.domain.model.Photo
import com.example.frontend.domain.model.Post
import com.example.frontend.domain.model.Video

data class PostDTO(
    val appUserId:Long,
    val appUserUsername:String,
    val postId:Long,
    val location:LocationDTO,
    val description:String,
    val numberOfLikes:Int,
    val numberOfComments:Int,
    val isLiked:Boolean,
    val photos:List<Photo>,
    //val videos:List<Video>
    val image: String,
    val date: String
)

fun PostDTO.toPost() : Post{
    return Post(
        postId = postId,
        description = description,
        appUserId = appUserId,
        appUserUsername = appUserUsername,
        numberOfLikes = numberOfLikes,
        numberOfComments = numberOfComments,
        isLiked = isLiked,
        photos = photos,
        //videos = videos,
        location = location.toLocation(),
        image = image,
        date = date
    )
}
